packages/shared/src/components/history/ReadingHistoryList.tsx

```diff
@@ -62,7 +64,12 @@ export default function ReadHistoryList({
 
         if (!currentDate || !isDateOnlyEqual(currentDate, date)) {
           currentDate = date;
-          dom.push(getDateGroup(date));
+          const dateToGroup = currentDate > now ? now : date;
```

Comment:
This will technically only happen for the first group "Today" since all other will fall under their regular groups?

Maybe just a naming wise we could use something like `mismatchedTimezoneGroup`.

Also maybe add this handling to separate if statement, I know it will add some code repetition but I think it will be more verbose in the long run. Like this it takes a minute to see what is actually going on.

Reply:
I think that's a great idea, I would rather have a separate scope to manage this specific use case than merge them together as you suggested. Will work that out. Thanks for pointing it out! 🚢